src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -334,6 +332,35 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         }
     }
 
+    // Method Description:
+    // - Sends the currently held text in the input buffer to the parent and
+    //   clears the input buffer and text block for the next round of input.
+    //   Then hides the text block control until the next time text received.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::_SendAndClearText()
+    {
+        // call event handler with data handled by parent
+        _compositionCompletedHandlers(_inputBuffer);

```

Comment:
Dumb q: the textblock is relaid-out between these events, right? When composition "completes" without starting, that is?

Reply:
Yup, the textblock actually seems to get relaid-out constantly between just about all events, such as TextUpdate, CompositionStarted, or CompositionCompleted events.